### DISCUSSION TITLE
FluidProperties module: locate out-of-range node during linear iterations

### DISCUSSION THREAD

**Discussion by garciapintado** [idaholab/moose#26551] [Q&A Modules: General] [answered]
Hello,
I have a simulation (based on PorousFlow) which uses the FluidProperties module and tabulated properties for pure water. At a point, it does not converge:

```
*** ERROR ***
The following error occurred in the object "tabulated_water95", of type "TabulatedBicubicFluidProperties".
Temperature 1643.76 is outside the range of tabulated temperature (273.17, 1400).

```
This value (1643.76) is way beyond the common values I have after convergence. I am wondering how to locate the node having this issue. Please, any suggestion?
Javier

**Comment by GiudGiud** [comment, marked as answer]
Hello

The best way is probably to output the fluid properties on FAILED and locate the area using the exodus output. 

You can use the FluidPropertiesMaterialPT for example to compute fluid properties as material properties. Then in that object you can set the output parameters to output to exodus

Note that returning an error for out of range is not what we should be doing here. We created the solution Invalid interface to avoid crashing simulations on out of range errors.

**Reply by garciapintado** [reply]
Hello Guillaume,
Many thanks! I had missed FluidPropertiesMaterialPT.

Also, I see that SolutionInvalidInterface is not in the version in our cluster. Will update it to see if/how that helps now

**Reply by GiudGiud** [reply]
I don't know if the interface is used in the fluid properties. 

But it should be used so we can change that if you see it s not

**Reply by GiudGiud** [reply]
Ok I checked, even in the latest moose it's still using a custom error behavior instead of the solutionInvalidityInterface.

However, you can pass the `error_on_out_of_bounds` parameter as false to let it continue with the solve and a bad temperature. Then you can use the material property output to see where it happened

**Comment by garciapintado** [comment]
Ah! Many thanks for checking!
OK. I'll try this...
 
On Monday, January 15, 2024 03:11 CET, Guillaume Giudicelli ***@***.***> wrote:
 
 
Ok I checked, even in the latest moose it's still using a custom error behavior instead of the solutionInvalidityInterface.
However, you can pass the error_on_out_of_bounds parameter as false to let it continue with the solve and a bad temperature. Then you can use the material property output to see where it happened
